apps/compiler/includes/functions.sh

```diff
@@ -208,9 +208,8 @@ function conf_layer() {
         if grep -qE "^$KEY" "$BASE" && ! grep -qE "^$KEY.*=.*$VAL" "$BASE"; then
           # Replace line
           # Prevent issues with shell quoting 
-          sed -i \
-            's,^'"$KEY"'.*,'"$KEY = $VAL$COMMENT"',g' \
-            "$BASE"
+          # Updated replace due to permissions creating config files across platforms. 
+          sed 's,^'"$KEY"'.*,'"$KEY = $VAL$COMMENT"',g' "$BASE" > tmp; cat tmp > "$BASE"; rm tmp
```

Comment:
`mktemp` creates a temporary filename that's guaranteed to be unique.

`cat` and redirection works, but they aren't needed here. `mv` makes more sense.

```suggestion
          TEMP="$(mktemp)"
          sed 's,^'"$KEY"'.*,'"$KEY = $VAL$COMMENT"',g' "$BASE" > "$TEMP" 
          mv "$TEMP" "$BASE"
```